CONTRIBUTING.md

```diff
@@ -89,6 +92,60 @@ Use the test scripts provided in the `ci/` directory to run tests for each compo
 ./ci/test_cub.sh -cxx g++ -std 14 -arch "70;75;80-virtual"
 ```
 
+### Using CMake Presets
+
+[CMake Presets](https://cmake.org/cmake/help/latest/manual/cmake-presets.7.html) are a set of configurations defined in a JSON file that specify project-wide build details for CMake. They provide a standardized and sharable way to configure, build, and test projects across different platforms and development environments. Presets are available from CMake version 3.19 and later.
+
+There are three kinds of Presets:
+
+- Configure Presets: specify options for the `cmake` command.
+
+- Builds Presets: specify options for the `cmake --build` command.
+
+- Test Presets: specify options for the `ctest` command.
+
+In CCCL we provide many presets to be used out of the box. You can find the complete list in the our corresponding [CMakePresets.json](./CMakePresets.json) file.
+
+#### Using CMake Presets via Command Line
+
+Once you have created your `build` directory and navigated into it, you can configure your project for a specific preset (e.g. `thrust-cpp11`) by executing the following command:
+
+```bash
+cmake --preset=thrust-cpp11 <path/to/cccl/root>
+```
+
+This command configures the project using the thrust-cpp11 preset. Please replace `<path/to/cccl/root>` with the actual path to the root directory of your CCCL project.
+
+Upon successful configuration, initiate the build process with:
+
+```bash
+cmake --build .
+```
+
```

Comment:
It would be awesome to have one more paragraph showing how to run the tests. Is it just executing `ctest` or do I have to build a special test target?

Reply:
will add it in a subsequent commit.